velox/exec/Merge.cpp

```diff
@@ -43,14 +45,20 @@ Merge::Merge(
         sortingKeys,
     const std::vector<core::SortOrder>& sortingOrders,
     const std::string& planNodeId,
-    const std::string& operatorType)
+    const std::string& operatorType,
+    const std::optional<common::SpillConfig>& spillConfig)
     : SourceOperator(
           driverCtx,
           std::move(outputType),
           operatorId,
           planNodeId,
-          operatorType),
-      outputBatchSize_{outputBatchRows()} {
+          operatorType,
+          spillConfig),
+      outputBatchSize_{outputBatchRows()},
+      maxNumMergeSources_{operatorCtx_->task()
```

Comment:
Shall we set maxNumMergeSources_ only if spill is enabled otherwise it is max?
```
maxNumMergeSources_{limit max}
```
Let's only set this in local merge operator?